packages/next/src/client/form.tsx

```diff
@@ -21,6 +21,19 @@ export type FormProps = {
    * - If `action` is a function, it will be called when the form is submitted. See the [React docs](https://react.dev/reference/react-dom/components/form#props) for more.
    */
   action: NonNullable<HTMLFormProps['action']>
+  /**
+   * Controls how the route specified by `action` is prefetched.
+   * Any `<Form />` that is in the viewport (initially or through scroll) will be prefetched.
+   * Prefetch can be disabled by passing `prefetch={false}`. Prefetching is only enabled in production.
+   *
+   * Options:
+   * - `null` (default): For statically generated pages, this will prefetch the full React Server Component data. For dynamic pages, this will prefetch up to the nearest route segment with a [`loading.js`](https://nextjs.org/docs/app/api-reference/file-conventions/loading) file. If there is no loading file, it will not fetch the full tree to avoid fetching too much data.
+   * - `true`: This will prefetch the full React Server Component data for all route segments, regardless of whether they contain a segment with `loading.js`.
+   * - `false`: This will not prefetch any data.
+   *
+   * @defaultValue `null`
+   */
+  prefetch?: boolean | null
```

Comment:
i'm wondering if we should disable the `true` option here and just make it `null | false`. Even with https://github.com/vercel/next.js/pull/68340, pages prefetched via `true` would never actually be used because the search params would differ. i'm not sure if there's a usecase where this makes sense. honestly it seem like the prefetch prop would mostly be used for *opting out*, perhaps if you have a bunch of Forms on one page

Reply:
changed it to `prefetch?: false | null`, `prefetch={true}` seems like it'll never do what people think, and if someone has an advanced usecase where they need that, they can do the prefetch on their own.